### DISCUSSION TITLE
Embedded 3D Models (OBJ, FBX) with Texture in Blazor Maui or Maui Mobile App

### DISCUSSION THREAD

**Discussion by TPitman-mindfiretechnology** [dotnet/maui#14834] [Q&A]
I have a mobile app that is mostly normal UI type elements that will share a lot of its pages with a website. The obvious best solution is to do the mobile and web apps as Blazor Maui apps.

There is one page in the app that is a couple of 3D models with textures. I need to be able to show them and move them around a little. Nothing fancy and not a lot of models.

I already have a test app I am working on that lets me mix the Blazor pages with a single xaml page.

I have already seen issue 129 and 196 and they don't really help because they talk mostly about game engines that pretty much take over the whole app. I could be wrong and if so, please clarify for me how I would do what I need to do.

I have also looked at MonoGame and it does the same full take over as far as I can tell.

I also looked at a nuget that wraps three.js (which I have used in another website app and worked great), but apparently JSInterop doesn't work in a Mobile Blazor Maui app. I got errors, tracked them down and found that not in the nuget package.

So does anyone have any ideas on how I can accomplish this?

**Comment by xtuzy** [comment]
use js is a simplest way, if you have abundant experience about 3d, you can create a 3d viewer base on https://github.com/xtuzy/Veldrid.Samples

**Reply by TPitman-mindfiretechnology** [reply]
I was just taking a look at this. Thank you for confirming it might be what I need.

**Comment by Xyncgas** [comment]
Embed cloud gaming, their enterprise solution let you display anything, 3D asset or playing GTA 5

**Reply by TPitman-mindfiretechnology** [reply]
I am not sure what you mean. If you are talking about using some kind of embedded link to a gaming server type thing I don't know if that will work. I need the 3D models and such to be local and embedded / controlled inside the app.

Can you give more details please?

**Reply by Xyncgas** [reply]
For example, Eve Online is a game, the offcial website let you play game if you are a paid member, in browser on the web. So that's the kind of technology I am talking about, I do not know what eve online use, but I was saying Nvidia's enterprise solution offers similar experience, similar to their cloud gaming product that let you play GTA5 in browser, it can render 3D asset or whatever and user can interact with it like dragging and it can work for many people on low performance device

**Reply by TPitman-mindfiretechnology** [reply]
Ah. Thank you for the thought, but that will not work for what I am doing. I just need to embed some simple 3D models and move them around.

The technology you are talking about is game streaming. This would require having a gaming server in order to just show a couple of simple 3D models. It might work, but it would be overkill for what I am trying to do.

Thank you again for the idea....

**Comment by GlacierFox** [comment]
I'm looking for this exact sort of functionality. I've got a set of 3D furniture designs I'd like to showcase in an app and allow the end user to simply inspect them by rotating them and perhaps zoom in on them. I haven't come across a good solution yet.

**Comment by shalindesigns2023** [comment]
Looking for [3D Furniture designer,](https://shalindesigns.com/furniture-design/) Here discover Top Furniture Designers in the UK, USA, India, Canada, and Germany, Specializing in Custom Furniture Joinery and Cabinet [Design Services.](https://www.reddit.com/user/shalindesign3D) Experience Exceptional Craftsmanship and Unleash the Beauty of Bespoke Furniture for Your Spaces.

**Reply by Xyncgas** [reply]
Sounds like something can be replaced by AI
